ignite/metrics/recsys/ndcg.py

```diff
@@ -0,0 +1,73 @@
+from typing import Callable, Optional, Sequence, Union
+
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric
+
+__all__ = ["NDCG"]
+
+
+class NDCG(Metric):
+    def __init__(
+        self,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+        k: Optional[int] = None,
+        log_base: Union[int, float] = 2,
+        exponential: bool = False,
+    ):
+
+        self.log_base = log_base
+        self.k = k
+        self.exponential = exponential
+        super(NDCG, self).__init__(output_transform=output_transform, device=device)
+
+    def _dcg_sample_scores(
```

Comment:
Let's put _dcg_sample_scores and _ndcg_sample_scores outside of the class as plain methods in ignite/metrics/recsys/ndcg.py